dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -583,9 +595,11 @@ public void reorderAction(final WorkflowAction action,
 	public WorkflowAction findAction(final String id, final User user) throws DotDataException, DotSecurityException {
 
 		final WorkflowAction action = workFlowFactory.findAction(id);
-		if (!APILocator.getPermissionAPI().doesUserHavePermission(action, PermissionAPI.PERMISSION_USE, user, true)) {
-			throw new DotSecurityException("User " + user + " cannot read action " + action.getName());
-		}
+
+		DotAssert.isTrue(
+				this.permissionAPI.doesUserHavePermission(action, PermissionAPI.PERMISSION_USE, user, true),
+				()-> "User " + user + " cannot read action " + action.getName(), DotSecurityException.class);

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " cannot read action " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "User " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)